### PR TITLE
VIH-9901 - new notify template for lip 3rd email 48 hour reminder

### PR DESCRIPTION
**Pull request by marcogagliardi**
[VIH-9901](https://tools.hmcts.net/jira/browse/VIH-9901)

### CODE REVIEW

**Review by shaed-parkar — COMMENTED** [review]
Few things to address to make it easier to maintain

File: SchedulerJobs/SchedulerJobs.Common/Configuration/FeatureToggles.cs [hunk 1]
```diff
@@ -11,6 +11,8 @@ public interface IFeatureToggles
         public bool StorePeopleIngestion();
         public bool WorkAllocationToggle();
         public bool ImportAllJudiciaryUsersToggle();
+        bool UsePostMay2023Template();
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply SchedulerJobs/SchedulerJobs.Common/Configuration/FeatureToggles.cs:14]: done

File: SchedulerJobs/SchedulerJobs.Services.UnitTests/AddNotificationRequestMapperTest.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    {
+        
+        var participantId = Guid.NewGuid();
+        var firstName = "firstname";
+        var lastName = "lastname";
+        var userName = "username";
+        var emailAddress = "contact@email.com";
+        var hearingDetails = new HearingDetailsResponse();
+        hearingDetails.Participants = new List<ParticipantResponse>();
+        ParticipantResponse participant1 = new ParticipantResponse();
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply SchedulerJobs/SchedulerJobs.Services.UnitTests/AddNotificationRequestMapperTest.cs:26]: done

File: SchedulerJobs/SchedulerJobs.Services.UnitTests/AddNotificationRequestMapperTest.cs [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        participant1.Username = userName;
+        participant1.UserRoleName = UserRoleNames.Individual;
+        hearingDetails.Participants.Add(participant1);
+        
+        var hearingId = Guid.NewGuid();    
+        var caseName = "random case name";
+        var caseNumber = "random case number";
+        
+
+        hearingDetails.Cases = new List<CaseResponse>();
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply SchedulerJobs/SchedulerJobs.Services.UnitTests/AddNotificationRequestMapperTest.cs:40]: done

File: SchedulerJobs/SchedulerJobs.Services.UnitTests/AddNotificationRequestMapperTest.cs [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        result.NotificationType.Should().Be(NotificationType.NewHearingReminderLipSingleDay);
+        result.MessageType.Should().Be(MessageType.Email);
+        result.Parameters.Should().BeEquivalentTo(parameters);
+    }
+    
+    [Test]
+    public void should_map_properties_for_notification_reminder_request_for_multiday_day_post_2023_toggle_true()
+    {
+        
+        var participantId = Guid.NewGuid();
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply SchedulerJobs/SchedulerJobs.Services.UnitTests/AddNotificationRequestMapperTest.cs:82]: refactored

File: SchedulerJobs/SchedulerJobs.Services/Mappers/AddNotificationRequestMapper.cs [hunk 5]
```diff
@@ -84,17 +97,25 @@ public static bool IsEjudge(ParticipantResponse participant)
             return (participant.ContactEmail.ToLower() == participant.Username.ToLower() && participant.Username.ToLower().Contains(Judiciary));
         }
 
-        private static Dictionary<string, string> InitCommonParameters(HearingDetailsResponse hearing)
+        private static Dictionary<string, string> InitCommonParameters(HearingDetailsResponse hearing, bool featureTogglePost2023 = false)
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply SchedulerJobs/SchedulerJobs.Services/Mappers/AddNotificationRequestMapper.cs:100]: I don't think you understood this. the toggle is used to decide to add parameters

File: SchedulerJobs/SchedulerJobs.Services/Mappers/AddNotificationRequestMapper.cs [hunk 6]
```diff
@@ -51,14 +57,21 @@ private static void AddParticipantParameters(ParticipantResponse participant, Di
...
-        private static NotificationType GetNotificationType(ParticipantResponse participant)
+        private static NotificationType GetNotificationType(ParticipantResponse participant, bool featureTogglePost2023 = false, bool isMultiDay = false)
         {
             NotificationType notificationType;
 
             switch (participant.UserRoleName)
             {
                 case UserRoleNames.Individual:
-                    notificationType = NotificationType.NewHearingReminderLIP;
+                    if (featureTogglePost2023)
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply SchedulerJobs/SchedulerJobs.Services/Mappers/AddNotificationRequestMapper.cs:67]: done

**Review by shaed-parkar — CHANGES_REQUESTED** [review]

File: SchedulerJobs/SchedulerJobs.Services/Mappers/AddNotificationRequestMapper.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply SchedulerJobs/SchedulerJobs.Services/Mappers/AddNotificationRequestMapper.cs:93]: I don't undesratnd

File: SchedulerJobs/SchedulerJobs/local.settings.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+  "IsEncrypted": false,
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply SchedulerJobs/SchedulerJobs/local.settings.json]: removed

**Review by shaed-parkar — APPROVED** [review]

**Review by shaed-parkar — APPROVED** [review]
The new notification api endpoints are available if you decide to swap over to it

**Review by komaldabhi1509 — APPROVED** [review]

**Pull request merged**